src/book/02-system/01-nubots/02-build-system.mdx

```diff
@@ -268,7 +268,49 @@ To run roles locally on your system use the `run` command of `b` as follows
 
 <Alert type="warning">
 
-Be sure that you are using the `generic` target for this (`./b target generic`), otherwise your role will likely crash (unless your CPU happens to be the same or newer than the 7th generation Core i7 that the robots are currently using).
+Important things to note when using the `run` command
+ - Be sure that you are using the `generic` target for this (`./b target generic`), otherwise your role will likely crash (unless your CPU happens to be the same or newer than the 7th generation Core i7 that the robots are currently using).
+ - When using the debugging utilities you should recompile your role after setting `CMAKE_BUILD_TYPE` to either `Debug` or `RelWithDebInfo` (`./b configure -i`) otherwise you will not get file or line number information in your stack traces.
+
+</Alert>
+
+The `run` command provides some options to help you debug your roles.
+
+| Subcommand | Description                                                                 |
+| ---------- | --------------------------------------------------------------------------- |
+| --asan     | Enables address sanitisation, a fast memory error detector.                 |
+| --gdb      | Runs selected role inside of GDB to enable debugging.                       |
+| --valgrind | Runs selected role inside of valgrind to enable detection of memory errors. |
+
+In order to use ASan you must recompile your role after setting `USE_ASAN` to `ON` in the cmake options (`./b configure -i`).
+
+Currently, using the `--asan` option adds an environment variable that will direct ASan to log all of its output to a file. This is done to ensure that even when you are running a curses-enabled role (e.g. `keyboardwalk`) you will always be able to recover the output from ASan (and it will also be clean from other log messages). There are a number of other run-time options you could set to tune the behaviour of ASan, they can be found [here](https://github.com/google/sanitizers/wiki/AddressSanitizerFlags). Just run your role as follows
```

Comment:
Maybe also specify where this file is if no path is specified (e.g. by default, the file is located in the NUbots directory in asan.log)

Reply:
If no file is specified it is logged to stderr